test/jdk/java/foreign/TestSharedAccess.java

```diff
@@ -138,6 +138,23 @@ public void testBadCloseWithPendingAcquire() throws InterruptedException {
         } //should fail here!
     }
 
+    @Test(expectedExceptions=IllegalStateException.class)
+    public void testBadCloseWithPendingAcquireBuffer() throws InterruptedException {
```

Comment:
I think this test can be improved to avoid the 5s execution time, plus a dangling thread hanging around for 500s.  With some latches we can ensure the main thread calls `segment.close()` while another other thread is within the `spliterator.forEachRemaining` (and further ensure the runnable task completes before the test completes).  I'll send a pull request.